File: gitdot-api/src/endpoint/review/create_review_comment.rs
use serde::{Deserialize, Serialize};
use uuid::Uuid;

use crate::{endpoint::Endpoint, resource::review::ReviewCommentResource};

pub struct CreateReviewComment;

impl Endpoint for CreateReviewComment {
    const PATH: &'static str = "/repository/{owner}/{repo}/review/{number}/comment";
    const METHOD: http::Method = http::Method::POST;

    type Request = CreateReviewCommentRequest;
    type Response = CreateReviewCommentResponse;
}

#[derive(ApiRequest, Debug, Serialize, Deserialize)]
pub struct CreateReviewCommentRequest {
    pub diff_id: Uuid,
    pub revision_id: Uuid,
    pub body: String,
    pub file_path: Option<String>,
    pub line_number_start: Option<i32>,
    pub line_number_end: Option<i32>,
    pub side: Option<String>,
}

pub type CreateReviewCommentResponse = ReviewCommentResource;

File: gitdot-web/packages/api/src/endpoint/review/create-review-comment.ts
import { z } from "zod";
import { ReviewCommentResource } from "../../resource";
import type { Endpoint } from "../endpoint";

export const CreateReviewCommentRequest = z.object({
  diff_id: z.string().uuid(),
  revision_id: z.string().uuid(),
  body: z.string(),
  file_path: z.string().optional(),
  line_number_start: z.number().int().optional(),
  line_number_end: z.number().int().optional(),
  start_character: z.number().int().optional(),
  end_character: z.number().int().optional(),
  side: z.string().optional(),
});
export type CreateReviewCommentRequest = z.infer<
  typeof CreateReviewCommentRequest
>;

export const CreateReviewCommentResponse = ReviewCommentResource;
export type CreateReviewCommentResponse = z.infer<
  typeof CreateReviewCommentResponse
>;

export const CreateReviewComment = {
  path: "/repository/{owner}/{repo}/review/{number}/comment",
  method: "POST",
  request: CreateReviewCommentRequest,
  response: CreateReviewCommentResponse,
} as const satisfies Endpoint;
export type CreateReviewComment = typeof CreateReviewComment;
